python/tests/test_lowlevel.py

```diff
@@ -303,6 +303,53 @@ def test_union_bad_args(self):
         with self.assertRaises(ValueError):
             tc.union(tc2, np.array([[1], [2]], dtype="int32"))
 
+    def test_idb_bad_args(self):
+        ts = msprime.simulate(10, random_seed=1)
+        tc = ts.tables.ll_tables
+        with self.assertRaises(TypeError):
+            tc.find_ibd()
+        for bad_samples in ["sdf", None, {}]:
+            with self.assertRaises(ValueError):
+                tc.find_ibd(bad_samples)
+        for not_enough_samples in [[], [0]]:
+            with self.assertRaises(_tskit.LibraryError):
+                tc.find_ibd(not_enough_samples)
+        for bad_float in ["sdf", None, {}]:
+            with self.assertRaises(TypeError):
+                tc.find_ibd([0, 1], min_length=bad_float)
+            with self.assertRaises(TypeError):
+                tc.find_ibd([0, 1], max_time=bad_float)
+
+    def test_idb_output_no_recomb(self):
+        ts = msprime.simulate(10, random_seed=1)
+        tc = ts.tables.ll_tables
+        segs = tc.find_ibd([0, 1, 2, 3])
```

Comment:
I've changed the interface here a bit. Rather than each value in the dictionary being a list of (left, right, node) tuples, I've changed to the values being dictionaries with keys "left", "right" and "node" which map to numpy arrays of the values. This is for efficiency at the C layer, and I think it'll end up being more useful later anyway when we use these for calculations.

Arguably we should make these values an ``IbdSegments`` class rather than plain dicts, but I'm not sure it matters much. 

What do you think? Will this break much of your existing Python code?

Reply:
So, just to be clear, this means that if a user wanted to see what IBD segments corresponds to sample pair `(u, v)`, they'd do something like this?
```
ibd_uv = ibd_output[(u, v)]
for i in range(len(ibd_uv)):
    print(ibd_uv['left'][i], ibd_uv['right'][i], ibd_uv['node'][i])
```
I think if this makes the C layer more efficient then we should do it, but it might be nice to have a Python function that converts this back into the more interpretable format.

> Will this break much of your existing Python code?

I could change the Python code, but it might be easier to just convert the existing Python output into this new format within the `verify_ibd` testing function.

> Arguably we should make these values an IbdSegments class rather than plain dicts, but I'm not sure it matters much.

Yeah, I was originally thinking of this output as a class of its own too, but I'm also not particularly attached to the idea!

Thanks for looking through this btw! I'll go over the other bits in more detail tomorrow.

Reply:
I think it's better to think of them as numpy arrays so we can do vector operations on them, so 

```python
ibd_uv = ibd_output[(u, v)]
left = ibd_uv["left"] # Although nicer if it was ``ibd_uv.left``
right =  ibd_uv["right"]
print("mean length = ", np.mean(right - left))
```

It's a classic struct-of-arrays vs array-of-structs thing. We've been tending to go for struct-of-arrays in tskit, so I'd tend to favour this approach for now. Let's see how it goes when we start working with some big tree sequences.